packages/tql/src/index.ts

```diff
@@ -0,0 +1,84 @@
+import { createQueryBuilder, isTemplateStringsArray } from './utils';
+import { TqlError } from './error';
+import type { CompiledQuery, Init } from './types';
+import {
+  TqlIdentifiers,
+  TqlList,
+  TqlNode,
+  TqlParameter,
+  TqlQuery,
+  TqlFragment,
+  TqlTemplateString,
+  TqlValues,
+  type TqlNodeType,
+  TqlSet,
+} from './nodes';
+import { build } from './build';
+
+export type * from './nodes';
+export type * from './types';
+export { PostgresDialect } from './dialects/postgres';
+
+export const init: Init = ({ dialect: Dialect }) => {
+  return {
+    query: (strings, ...values): CompiledQuery => {
+      const query = parseTemplate(TqlQuery, strings, values);
+      const qb = createQueryBuilder();
+      const d = new Dialect(qb.appendToQuery, qb.appendToParams);
+      const preprocessed = d.preprocess(query);
+      build(d, preprocessed);
+      return d.postprocess(qb.query, qb.params);
+    },
+    fragment: (strings, ...values) =>
+      parseTemplate(TqlFragment, strings, values),
+    identifiers: (ids) => new TqlIdentifiers(ids),
+    list: (vals) => new TqlList(vals),
+    values: (entries) => new TqlValues(entries),
+    set: (entries) => new TqlSet(entries),
```

Comment:
It's unfortunate that `list` and `set` in this context here are very different than your typical list/set and might confuse users (e.g. `Array` vs `Set`). Is it worth trying to separate the concepts of list/set and list/sql-update-set? Maybe `list` becomes `join` or `array`? (`join` is actually pretty common).

Reply:
This works really well in the `SET` case:

```sql
UPDATE users SET name = $1 WHERE user_id = $2;
```

becomes:

```ts
query`UPDATE users ${SET(updatedUser)}`;
```

which "reads" the same, meaning it'll be easy to remember / write.

Reply:
Implemented this in #507 for an example.